### ISSUE TITLE
Questions on Utilizing COBRA for Megaphage and Plasmid Analysis

### ISSUE THREAD

**Issue by ucaswhw** [linxingchen/cobra#49]
Thank you very much for providing these tools. I have two questions I would like to consult with you. My research focuses on the ecological functions and evolution of bacteriophages.
	1.	Can the high-quality megaphage sequences recovered through COBRA be analyzed together with non-megaphage contigs, or should they be analyzed post-binning?
	2.	Is it possible to use COBRA to recover or extend plasmid fragments?

I would greatly appreciate it if you could provide answers to these questions. Thank you once again for your kind assistance!

**Comment by linxingchen** [comment]
Hi, thank you for your questions.
1. COBRA does not check if a query comes from a megaphage or a non-megaphage. 
2. Yes, it is doable.
Please let me know if you have more questions.

**Comment by linxingchen** [comment]
Please keep in mind all queries (-q) and contigs/scaffolds (-f) should be original from the assembly without cutting or any other modifications.

**Comment by ucaswhw** [comment]
Thank you for your response. Regarding the first question, my earlier phrasing may not have been clear. Can the high-quality viral contigs recovered using COBRA be analyzed downstream together with the remaining contigs that could not be recovered? Would such an approach introduce systematic biases?

**Comment by ucaswhw** [comment]
Yes, I understand that all queries (-q) and contigs/scaffolds (-f) should originate from the original assembly.

**Comment by linxingchen** [comment]
good question.

it is fine. COBRA will generate a new fasta file, which includes all the unique joined sequences and other original sequences that have not been joined, for subsequent analyses.

**Comment by ucaswhw** [comment]
Got it, thank you so much—I truly appreciate your help!

**Comment by linxingchen** [comment]
if your queries are predicted viral contigs/scaffolds, and you want to focus on them for the subsequent analyses, you could directly combine the five fasta files of the five categories.

**Comment by ucaswhw** [comment]
I understand.

**Comment by ucaswhw** [comment]
Hello Professor Chen, another question: You mentioned that all queries (-q) and contigs/scaffolds (-f) should be original from the assembly without cutting or any other modifications. However, If I have generated scaffolds using MetaSPAdes and produced some viral scaffolds with MetaVirusSPAdes, is it appropriate to merge these datasets and then use a viral classifier to select those sequences identified as viral, subsequently using them as input for COBRA?

**Comment by linxingchen** [comment]
Hi, I have not tested MetaVirusSPAdes, but please remember that -q and -f contigs/scaffolds should come from the same assembly. I hope this helps.
